File: backend/src/main/java/com/prepunite/resource/entity/Resource.java
package com.prepunite.resource.entity;

import com.prepunite.common.entity.BaseEntity;
import com.prepunite.common.enums.ResourceCategory;
import jakarta.persistence.*;
import lombok.*;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "resources")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Resource extends BaseEntity {

    @Column(nullable = false, length = 500)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String url;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ResourceCategory category;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "thumbnail_url", columnDefinition = "TEXT")
    private String thumbnailUrl;

    @Column(name = "is_verified", nullable = false)
    @Builder.Default
    private Boolean isVerified = true;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    @Column(name = "view_count", nullable = false)
    @Builder.Default
    private Integer viewCount = 0;

    @ElementCollection
    @CollectionTable(name = "resource_tags", joinColumns = @JoinColumn(name = "resource_id"))
    @Column(name = "tag")
    @Builder.Default
    private Set<String> tags = new HashSet<>();
}

File: frontend/src/types/resource.ts
export type ResourceCategory =
  | 'YOUTUBE'
  | 'NOTES'
  | 'PDF'
  | 'CHEAT_SHEET'
  | 'ARTICLE'
  | 'PLAYLIST'
  | 'DOCUMENTATION'
  | 'BOOK'
  | 'PRACTICE_WEBSITE';

export interface Resource {
  id: string;
  title: string;
  url: string;
  category: ResourceCategory;
  description?: string;
  thumbnailUrl?: string;
  isVerified: boolean;
  viewCount: number;
  tags?: string[];
  createdAt: string;
}
